go/engine/revoke.go

```diff
@@ -122,37 +126,104 @@ func (e *RevokeEngine) Run(ctx *Context) error {
 		ctx.LogUI.Info("  %s", kid)
 	}
 
-	ska := libkb.SecretKeyArg{
-		Me:      me,
-		KeyType: libkb.DeviceSigningKeyType,
-	}
-	sigKey, err := e.G().Keyrings.GetSecretKeyWithPrompt(ctx.SecretKeyPromptArg(ska, "to revoke another key"))
+	sigKey, encKey, err := e.getDeviceSecretKeys(ctx, me)
 	if err != nil {
 		return err
 	}
-	if err = sigKey.CheckSecretKey(); err != nil {
-		return err
+
+	if e.G().Env.GetEnableSharedDH() {
```

Comment:
We're going to need to run this code from outside this engine when a user is upgraded from to using DH from legacy.